lib/sensu/api/process.rb

```diff
@@ -335,18 +335,9 @@ def resolve_event(event_json)
             clients.each_with_index do |client_name, index|
               settings.redis.get("client:#{client_name}") do |client_json|
                 unless client_json.nil?
-                  begin
-                    response << MultiJson.load(client_json)
-                  rescue MultiJson::ParseError => error
```

Comment:
Only observed JSON parsing errors are when `client_json` was `nil`, before we added the conditional. AFAIK Redis does not support partial writes and the Redis client library only calls the callback after a full read.